### PR TITLE
[REF] iot_box_image: move image build tools from pos to module

### PR DESCRIPTION
**Pull request by loouis-t**
A long long time ago, the IoT Box was called the PoS Box. As a result, all the tools to build an image for it were placed in the `point_of_sale` module.

Now, these tools have their own module.

### CODE REVIEW

**Comment by loouis-t** [comment]
@odoo/rd-security could you have a look at this please?

## Context

We just moved all files related to building an IoT Box image from the `point_of_sale` module to a new `iot_box_module`.

## Security issues

multiple `subprocess.check_call(['sudo', 'bash', '-c', '. /home/pi/odoo/addons/iot_box_image/configuration/upgrade.sh; cleanup'])` where we only changed the path.

**Review by lse-odoo — COMMENTED** [review]
would/could/should not change the git sparse-checkout file of the IoT box ? :thinking: 
https://github.com/odoo/odoo/blob/c551b9e850bebb33a6d434cf6538e182c60d4db0/addons/point_of_sale/tools/posbox/posbox_create_image.sh#L67

**Comment by loouis-t** [comment]
> would/could/should not change the git sparse-checkout file of the IoT box ? 🤔
> 
> https://github.com/odoo/odoo/blob/c551b9e850bebb33a6d434cf6538e182c60d4db0/addons/point_of_sale/tools/posbox/posbox_create_image.sh#L67

it's updated [here](https://github.com/odoo/odoo/pull/186740/files#diff-4552bd40eedcf37860191309f18304a7e1eef50ea8175f5ba550d22bb690b6faR67), and [here](https://github.com/odoo/odoo/pull/186740/files#diff-1f4219946f3d8b40b98123cd292cfad310025f647bb9b44c83c25fb3c67f422dR13). If the old path is still present in the sparse-checkout, git will just ignore it.

**Review by xmo-odoo — COMMENTED** [review]
@robodoo override=ci/security

File: addons/iot_box_image/configuration/connect_to_wifi.sh
Comment (Ysoroko) [inline addons/iot_box_image/configuration/connect_to_wifi.sh:1]: Wifi works as before

**Review by Ysoroko — APPROVED** [review]
Everything works as expected, just need to double check a connection to the db in v17

**Comment by Ysoroko** [comment]
@robodoo r+

**Pull request closed without merging**